### ISSUE TITLE
authentication header is case sensitive.

### ISSUE THREAD

**Issue by erahhal** [python-caldav/caldav#458]
Extraction of `WWW-Authenticate` header appears to be case sensitive. Headers are not case sensitive according to the HTTP standard, and Caddy sends `www-authenticate`, which can't be seen by the davclient, e.g in [this line](https://github.com/python-caldav/caldav/blob/master/caldav/davclient.py#L732).

**Comment by tobixen** [comment]
Sorry the noise, should try to remember not to do any commits before getting my morning cafffeine next time :-)

So the requests library has a CaseInsensitiveDict which is meant to be used exactly for this purpose.  It was used in parts of the code, but not consistently.

Could you please try to use either the issue458 branch or the v1.x-branch and see if that solves your problem?  (hm, lack of morning caffeine again ... I shouldn't check things into the v1.x-branch before it has been merged to master)

**Comment by erahhal** [comment]
I'm not actually using it directly. Was trying to use another package called Errands and it kept failing to authenticate so I dug into it and found out it was using this library.  Would need to get that app running in a local dev environment first.

**Comment by tobixen** [comment]
I should probably try to write up some unit tests that mocks the authentication, but ... I'm not up for it today at least.

If you would have the possibility to create a test account for me on your caldav server it would be great, then I could run the test code towards it.

(The authentication part ought to be solved by the Requests library, it's silly that I need to do all this low-level http stuff in the caldav library.  Now that we're moving away from requests, perhaps this code can be discarded completely)

**Comment by tobixen** [comment]
Correct.  The headers returned from the requests library is a `CaseInsensitiveDict`, hence `r.headers['WWW-Authenticate']` will always work, no matter if the server returns the header in lower case or upper case (but I should fix that `DAVClients.headers` is a `CaseInsensitiveDict` anyway).

**Comment by tobixen** [comment]
There should of course have been some test code here ... but it would be required with quite some mocking up on a quite low level.  I assume the problem has been fixed, so I will close this issue to remove the clutter.
